cumulus/tasks/pdr-status-check/index.js

```diff
@@ -141,34 +157,28 @@ function buildOutput(event, groupedExecutions) {
  */
 async function checkPdrStatuses(event) {
   const runningExecutionArns = event.input.running || [];
+  const concurrencyLimit = process.env.CONCURRENCY || 10;
+  const limit = pLimit(concurrencyLimit);
 
-  const executions = [];
-  for (const executionArn of runningExecutionArns) {
-    try {
-      const execution = await aws.sfn().describeExecution({ executionArn }).promise();
-      executions.push(execution);
-    }
-    catch (e) {
-      // it's ok if a execution is still in the queue and has not be executed
-      if (e.code === 'ExecutionDoesNotExist') {
-        executions.push({ executionArn: executionArn, status: 'RUNNING' });
-      }
-      else throw e;
-    }
-  }
+  const promisedExecutionDescriptions = runningExecutionArns.map((executionArn) =>
+    limit(() => describeExecutionStatus(executionArn)));
```

Comment:
Is my understanding correct that this limits concurrency to 10 such that if there are 100 running executions, it would only fetch 10 statuses at a time?

Related question - is there any retry functionality built into this? Like if one of the connections fails we retry a request for the execution arn? Not required for this fix but just curious.

Reply:
Yes, only 10 sfn.DescribeExecution can be run at a time.  
It's not retried, if any sfn.DescribeExecution fails, the lambda function will fail.  I think we could configure the lambda function to be retried in case of failure, although I haven't done one.